Agent: Variable overflow in fill_cs_params

VladyslavTupikin opened the pull request:
Function set_tx_power_limit break the logic for
tx_power variable when multiple tx_pow_limit on 100
and assign it to the m_radio_info.tx_power, because in
other cases as example in the function fill_cs_params
tx_power has type int8_t, and it triggers overflow when
value in tx_power greater than 127.

Solution - remove multiplication from set_tx_power and
add type conversion from int to int8_t in fill_cs_params
for tx_power.

Commit:
Agent: Variable overflow in fill_cs_params

Function set_tx_power_limit break the logic for
tx_power variable when multiple tx_pow_limit on 100
and assign it to the m_radio_info.tx_power, because in
other cases as example in the function fill_cs_params
tx_power has type int8_t, and it triggers overflow when
value in tx_power greater than 127.

Solution - remove multiplication from set_tx_power and
add type conversion from int to int8_t in fill_cs_params
for tx_power.

Signed-off-by: Vladyslav Tupikin <v.tupikin@inango-systems.com>

tomereli commented on agent/src/beerocks/slave/ap_manager_thread.cpp:
```diff
@@ -983,7 +983,7 @@ bool ap_manager_thread::handle_cmdu(Socket *sd, ieee1905_1::CmduMessageRx &cmdu_
 void ap_manager_thread::fill_cs_params(beerocks_message::sApChannelSwitch &params)
 {
-    params.tx_power  = ap_wlan_hal->get_radio_info().tx_power;
+    params.tx_power  = (int8_t)ap_wlan_hal->get_radio_info().tx_power;
```
```suggestion
    params.tx_power  = static_cast<int8_t>(ap_wlan_hal->get_radio_info().tx_power);
```

arnout replied:
Actually, for conversions between scalar types, it [should use `T{e}` construction notation](https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#Res-construct).
```suggestion
    params.tx_power  = int8_t{ap_wlan_hal->get_radio_info().tx_power};
```

However, this particular one is redundant, because `params.tx_power` is already an int8_t. For the other two it's needed though.

tomereli commented on common/beerocks/bwl/dummy/ap_wlan_hal_dummy.cpp:
```diff
@@ -327,8 +327,8 @@ bool ap_wlan_hal_dummy::read_supported_channels() { return true; }
 {
-    LOG(TRACE) << __func__ << " setting power limit: " << tx_pow_limit * 100 << " mBm";
-    m_radio_info.tx_power = tx_pow_limit * 100;
+    LOG(TRACE) << __func__ << " setting power limit: " << tx_pow_limit;
```
```suggestion
    LOG(TRACE) << __func__ << " setting power limit: " << tx_pow_limit << " dBM";
```

arnout replied:
```suggestion
    LOG(TRACE) << __func__ << " setting power limit: " << int8_t{tx_pow_limit} << " dBm";
```

arnout replied:
This apparently doesn't work...

tomereli reviewed (approved):
Please fix the C style cast to a static cast.
In addition, the commit message only says there is an overflow due to the fact we multiply the tx power value by 100, which is true, but doesn't explain why this is done - the reason is that the actual NL command expects the tx power in mBm which is why we multiply this by 100 in the actual implementation, but when we read the value from hostapd, it is in dBM so we don't need to divide by 100.
In the dummy implementation though, there is a single API which already sets the dBM value, and it was wrong to multiply by 100 in the first place.

Other than this, LGTM!

arnout commented:
Forgot to mention: nice catch!

tomereli commented:
@VladyslavTupikin please fix the comments, then add the `ready for merge` tag so that `mergify` will merge this PR automatically.

arnout commented:
@VladyslavTupikin *Not* with fixup commits if it's intended to be merged. I removed the don't merge tag again since I'm not sure if mergify takes into account the DCO check.

Commit:
Agent: Variable overflow in fill_cs_params

Type conversation in current changes done by C-style
type conversation.

Changed C-style type conversation (int8_t) on C++-style
type conversation  static_cast<int8_t>().

Signed-off-by: Vladyslav Tupikin <v.tupikin@inango-systems.com>

arnout commented:
If you have the time, it would be nice if you could also add something to test_flows.py that captures this error. Or is that covered by #965?
